client/service_create.go

```diff
@@ -18,13 +20,28 @@ func (cli *Client) ServiceCreate(ctx context.Context, service swarm.ServiceSpec,
 		}
 	}
 
+	// Call the /manifest endpoint to retrieve manifest information
+	var manifestInspect registrytypes.ManifestInspect
+	manifestResp, manifestErr := cli.get(ctx, "/images/"+service.TaskTemplate.ContainerSpec.Image+"/manifest", url.Values{}, headers)
```

Comment:
You mean the image already contains a digest? Then the digest will be used to retrieve the manifest, and platform information added (this is not part of this PR, and will come later in a separate PR). Without the platform information stuff, we could just skip this call altogether.